show/plugins/barefoot.py

```diff
@@ -37,9 +39,10 @@ def profile():
     
     # Print profile list
     click.echo('Available profile(s):')
-    subprocess.run('docker exec -it syncd find /opt/bfn -mindepth 1 '
-        r'-maxdepth 1 -type d -name install_\*_profile ' + opts + '| sed '
-        r's%/opt/bfn/install_\\\(.\*\\\)_profile%\\1%', shell=True)
+    cmd0 = ['docker', 'exec', '-it', 'syncd', 'find', '/opt/bfn', '-mindepth', '1',\
+            r'-maxdepth', '1', '-type', 'd', '-name', 'install_\*_profile', r'{}' % opts]
```

Comment:
>r'{}' % opts [](http://example.com/codeflow?start=76&length=12)

What the diff if using `opts` directly?

Reply:
`r''` is to bypass escaping special characters. Using `opts` directly will escape special characters.

Reply:
Could you be more specific?, `r` is applied to `{}` string only.

Reply:
for example, `r'\n'` will treated the string containing 2 characters: backslash `\` and letter `n`. But '\n' will be treated as a newline character.